biogtr/config.py

```diff
@@ -0,0 +1,228 @@
+# to implement - config class that handles getters/setters
+"""Data structures for handling config parsing."""
+import hydra
+import sys
+import ast
+import torch
+import pytorch_lightning as pl
+from biogtr.models.global_tracking_transformer import GlobalTrackingTransformer
+from biogtr.training.losses import AssoLoss
+from biogtr.datasets.sleap_dataset import SleapDataset
+from biogtr.datasets.microscopy_dataset import MicroscopyDataset
+from omegaconf import DictConfig, OmegaConf
+from typing import Union, Iterable
+
+
+class Config:
```

Comment:
We should have a test for the config getters/setters. However, having codecov target at 100% is probably overkill. We should maybe have a threshold of 20% (will add a codecov.yml to deal with this.)

Reply:
ok will add tests for config